Ref.Windows/ViewModels/CopyReferenceDialogViewModel.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Text;
+
+namespace Polsys.Ref.ViewModels
+{
+    /// <summary>
+    /// The view model for the Copy Reference dialog.
+    /// </summary>
+    internal class CopyReferenceDialogViewModel : ViewModelBase
+    {
+        /// <summary>
+        /// Gets the full citation text.
+        /// </summary>
+        public string Citation
+        {
+            get { return GenerateCitation(); }
+        }
+
+        /// <summary>
+        /// Gets or sets the citation style.
+        /// </summary>
+        public CitationStyle CitationStyle
+        {
+            get { return _citationStyle; }
+            set
+            {
+                if (_citationStyle != value)
+                {
+                    _citationStyle = value;
+                    NotifyPropertyChanged(nameof(CitationStyle));
+                    NotifyPropertyChanged(nameof(Citation));
+                }
+            }
+        }
+        private CitationStyle _citationStyle;
+
+        /// <summary>
+        /// Gets or sets the citation markup type.
+        /// </summary>
+        public ReferenceOutputType OutputType
+        {
+            get { return _outputType; }
+            set
+            {
+                if (_outputType != value)
+                {
+                    _outputType = value;
+                    NotifyPropertyChanged(nameof(OutputType));
+                    NotifyPropertyChanged(nameof(Citation));
+                }
+            }
+        }
+        private ReferenceOutputType _outputType;
+
+        private PublicationViewModelBase _entry;
+
+        /// <summary>
+        /// Creates a new view model instance with the specified entry.
+        /// </summary>
+        /// <param name="entry">The entry to display reference for.</param>
+        public CopyReferenceDialogViewModel(PublicationViewModelBase entry)
+        {
+            _entry = entry;
+        }
+
+        private string GenerateCitation()
+        {
+            switch (CitationStyle)
+            {
+                case CitationStyle.Apa:
+                    return GenerateApa();
+                case CitationStyle.Chicago:
+                    return GenerateChicago();
+                default:
+                    throw new ArgumentException("Unknown citation style");
+
+            }
+        }
+
+        private string GenerateApa()
+        {
+            // Already has some additional capacity
+            var result = new StringBuilder(128);
+
+            if (_entry is ArticleViewModel)
+            {
+                var article = _entry as ArticleViewModel;
+
+                WriteAuthors(result, article.Author);
+                result.Append(" (");
+                result.Append(article.Year);
+                result.Append("). ");
```

Comment:
This is quite ugly but I don't think it's worth generalizing too much. Of course, if there are lots of styles in the future, this can be refactored.